### PR TITLE
Changes to introductory text.

### PR DESCRIPTION
**Pull request by MarkoM-dot**
closes https://github.com/rust-unofficial/patterns/discussions/297

### CODE REVIEW

**Review by MarcoIeni — APPROVED** [review]

**Comment by MarcoIeni** [comment]
Sounds good to me! Let's see if someone else has other feedback.

**Review by MarcoIeni — APPROVED** [review]

File: intro.md [hunk 1]
```diff
@@ -7,10 +7,17 @@ If you are interested in contributing to this book, check out the
...
-A program can be viewed as a solution to a problem.
-It can also be viewed as a collection of solutions to many problems.
-All of these solutions work together to solve a bigger problem.
+In software development, we often come across problems that share
+similarities regardless of the environment they appear in. Although the
+implementation details are crucial to solve the task at hand, we may
+abstract from these particularities to find the common practices that
+are generically applicable.
+
+Design patterns are a collection of reusable tried and tested solutions to
```
Comment (MarcoIeni) [inline intro.md:16]:
```suggestion
Design patterns are a collection of reusable and tested solutions to
```
I think tried is redundant. Tested should be enough.

**Review by MarcoIeni — CHANGES_REQUESTED** [review]

File: intro.md (hunk 1 above)
Comment (MarkoM-dot) [reply intro.md]: I like it! Thank you.

**Review by MarcoIeni — APPROVED** [review]

**Comment by MarcoIeni** [comment]
Thanks!

**Pull request merged**